### DISCUSSION TITLE
Rio segfaults at the kernel level on Arch Linux

### DISCUSSION THREAD

**Discussion by theoparis** [raphamorim/rio#834] [Polls]
```
[  158.163540] rio[3627]: segfault at 40 ip 00007d7dfce163c4 sp 00007ffd27e7b058 error 4 in libvulkan_radeon.so[2163c4,7d7dfcc0e00
0+4f3000] likely on CPU 5 (core 5, socket 0)
[  158.163549] Code: 0f 7e 43 40 48 8d 0d db b3 49 00 f3 0f 7e 0d 9b d7 32 00 48 89 48 08 66 0f fe c1 66 0f d6 43 40 5b 41 5c 5d c
3 90 f3 0f 1e fa <8b> 47 40 85 c0 74 3f 48 8b 57 08 48 85 f6 75 3c 8b 4f 20 48 89 d6
[  395.613438] amdgpu 0000:2f:00.0: amdgpu: bo 000000006ffb8fc4 va 0x0800100742-0x0800100755 conflict with 0x0800100742-0x08001007
56
[  395.633437] ------------[ cut here ]------------
```

I'm not sure if this is a mesa, rio, or kernel bug so I decided to just report it here for now. I have mesa 24.2.7 and a 6.12.1 Linux kernel. It worked fine the other day before I updated my system... Hyprland and xterm both ran fine, but running rio causes no output. It opens the window and the immediately crashes with the above error in dmesg. There is no indication of a user level segfault because it exits with a status code of 0....

**Comment by theoparis** [comment]
It turns out I was missing my shell executable at ~/.cargo/bin/nu. However it seems weird that it would segfault like that

**Comment by raphamorim** [comment]
Hey @theoparis thanks for the issue!

Oh yea, it's weird. Rio often return an error saying the "file" (in case binary file) was not found.

Was that error on release target?

I will be closing this issue for now since you figured out the cause but I will move to discussion to further investigation
